tests/test_local_network_access_manager.cpp

```diff
@@ -0,0 +1,231 @@
+
+        std::unique_ptr<QNetworkReply> reply{manager.sendCustomRequest(request, verb, data)};
+
+        QObject::connect(reply.get(), &QNetworkReply::finished, &event_loop, &QEventLoop::quit);
+        QObject::connect(&download_timeout, &QTimer::timeout, [&] {
+            download_timeout.stop();
+            reply->abort();
+        });
+
+        download_timeout.start();
+        event_loop.exec();
+
+        return reply;
+    }
+
+    mp::NetworkAccessManager manager;
+    QLocalServer test_server;
+    mpt::TempDir temp_dir;
+    QString socket_path;
+    QUrl base_url;
+    QEventLoop event_loop;
+    QTimer download_timeout;
+};
+
+struct HTTPErrorsTestSuite : LocalNetworkAccessManager, WithParamInterface<HTTPErrorParamType>
+{
+};
+
+const std::vector<HTTPErrorParamType> http_error_suite_inputs{
+    {"HTTP/1.1 400 Bad Request\r\n\r\n", QNetworkReply::ProtocolInvalidOperationError},
+    {"HTTP/1.1 401 Authorization Required\r\n\r\n", QNetworkReply::AuthenticationRequiredError},
+    {"HTTP/1.1 403 Access Denied\r\n\r\n", QNetworkReply::ContentAccessDenied},
+    {"HTTP/1.1 404 Not Found\r\n\r\n", QNetworkReply::ContentNotFoundError},
+    {"HTTP/1.1 409 Resource Conflict\r\n\r\n", QNetworkReply::ContentConflictError},
+    {"HTTP/1.1 500 Internal Server Error\r\n\r\n", QNetworkReply::InternalServerError},
+    {"HTTP/1.1 501 Unknown Server Error\r\n\r\n", QNetworkReply::UnknownServerError},
+    {"HTTP/1.1 412 Precondition Failed\r\n\r\n", QNetworkReply::UnknownContentError}};
+} // namespace
+
+TEST_F(LocalNetworkAccessManager, no_error_returns_good_reply)
+{
+    QByteArray http_response;
+    http_response += "HTTP/1.1 200 OK\r\n";
+    http_response += "Content-Type: text/plain\r\n";
+    http_response += "\r\n";
+
+    local_socket_server_handler(http_response);
+
+    auto reply = handle_request(base_url, "GET");
+
+    EXPECT_THAT(reply->error(), Eq(QNetworkReply::NoError));
+}
+
+TEST_F(LocalNetworkAccessManager, reads_expected_data_not_chunked)
+{
+    QByteArray reply_data{"Hello"};
+
+    QByteArray http_response;
+    http_response += "HTTP/1.1 200 OK\r\n";
+    http_response += "Content-Type: text/plain\r\n";
+    http_response += "Content-Length: 5\r\n";
+    http_response += "\r\n";
+    http_response += reply_data;
+    http_response += "\r\n";
+
+    local_socket_server_handler(http_response);
+
+    auto reply = handle_request(base_url, "GET");
+
+    ASSERT_THAT(reply->error(), Eq(QNetworkReply::NoError));
+
+    auto data = reply->readAll();
+
+    EXPECT_THAT(data, Eq(reply_data));
+}
+
+TEST_F(LocalNetworkAccessManager, reads_expected_data_chunked)
+{
+    QByteArray reply_data{"What's up?"};
+
+    QByteArray http_response;
+    http_response += "HTTP/1.1 200 OK\r\n";
+    http_response += "Content-Type: text/plain\r\n";
+    http_response += "Content-Length: 5\r\n";
+    http_response += "Transfer-Encoding: chunked\r\n";
+    http_response += "\r\n";
+    http_response += "5\r\n";
```

Comment:
This is 53, no? So your reply is a bit malformed? ;P

Reply:
I'm not really following you.  The data is 5 bytes (`Hello`), so the chunked data size is `5` here.

Reply:
It's actually "What's up?", no? And I misunderstood what you meant by "in hex" above, so this should actually be 9, right?

Reply:
Oh, I was looking at the wrong test :grin:   Will fix.

Reply:
Fixed